core/src/progress_map.rs

```diff
@@ -104,6 +104,27 @@ impl ReplaySlotStats {
                 i64
             ),
         );
+
+        let mut per_pubkey_timings: Vec<_> = self
+            .execute_timings
+            .details
+            .per_program_timings
+            .iter()
+            .collect();
+        per_pubkey_timings.sort_by(|a, b| b.1.cmp(&a.1));
+        let total: u64 = per_pubkey_timings.iter().map(|a| a.1).sum();
+        for (pubkey, time) in per_pubkey_timings.iter().take(5) {
```

Comment:
Perhaps to remove `take(5)` when utilize this change to feed cost model?

Reply:
Well this is just to report metrics for the top 5 costly programs. So I think we would leave this in. There would just be another code-path to update the cost model with these values.

Reply:
right, makes sense, this function just reports stats. Will add code at replay_stage to read execute_timings for cost model